src/coreclr/src/jit/lower.cpp

```diff
@@ -4529,7 +4529,16 @@ GenTree* Lowering::LowerVirtualStubCall(GenTreeCall* call)
         }
         else
         {
-            result = Ind(addr);
+
+            bool shouldOptimizeVirtualStubCall = false;
```

Comment:
I think this would be a good place to describe exactly what's being optimized and how, and to touch on how  this is supported downstream (i.e. in register allocation and codegen). Also, at first I didn't grasp the fact that the tailcall exception is handled in codegen due to the fact that it checks for a non-null control expr before it goes down the "optimized" path.

Reply:
Sure, I will add a comment. 

>  I didn't grasp the fact that the tailcall exception is handled in codegen due to the fact that it checks for a non-null control expr before it goes down the "optimized" path.

That was the reason the tests were failing and I realized that I should be not doing the optimization in presence of tailcall.